From 38ac8101c481972b42374e2ed29d860712eae011 Mon Sep 17 00:00:00 2001
From: Mauro Rossi <issor.oruam@gmail.com>
Date: Sat, 10 Jul 2021 21:12:38 +0200
Subject: [PATCH 3/3] libstagefright: Extended media support via FFMPEG (s-x86
 v2)

Fixes the following building error:

In file included from frameworks/av/media/libstagefright/flac/dec/FLACDecoder.cpp:26:
In file included from frameworks/av/media/libstagefright/include/media/stagefright/MediaDefs.h:29:
frameworks/av/media/libstagefright/foundation/include/media/stagefright/foundation/MediaDefs.h:156:13: error: enumeration values
 'kAudioEncodingPcm24bitPacked' and 'kAudioEncodingPcm32bit' not handled in switch [-Werror,-Wswitch]
    switch (encoding) {
            ^
1 error generated.

Change-Id: I43a677683ff51d2138883346dd80bd0d19d765c9
---
 .../include/media/stagefright/foundation/MediaDefs.h          | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/media/libstagefright/foundation/include/media/stagefright/foundation/MediaDefs.h b/media/libstagefright/foundation/include/media/stagefright/foundation/MediaDefs.h
index 3b40067379..fec1498e94 100644
--- a/media/libstagefright/foundation/include/media/stagefright/foundation/MediaDefs.h
+++ b/media/libstagefright/foundation/include/media/stagefright/foundation/MediaDefs.h
@@ -161,6 +161,10 @@ static inline int32_t audioEncodingToBits(AudioEncoding encoding) {
             return 8;
         case kAudioEncodingPcm16bit:
             return 16;
+        case kAudioEncodingPcm24bitPacked:
+            return 24;
+        case kAudioEncodingPcm32bit:
+            return 32;
         case kAudioEncodingPcmFloat:
             return 32;
     }
-- 
2.25.1

